### PR TITLE
G3Writer: release the GIL during IO

### PR DESCRIPTION
**Pull request by mhasself**
This will make life easier in the Simons Observatory Smurf application.

Thanks for the GenerateBlobs api!

### CODE REVIEW

File: core/src/G3Writer.cxx [hunk 1]
```diff
@@ -34,6 +34,16 @@ G3Writer::~G3Writer()
 
 void G3Writer::Process(G3FramePtr frame, std::deque<G3FramePtr> &out)
 {
+        // If in Python context, release the GIL while writing out.
+        // Note we must force serialization of the frames, first,
+        // because in some cases serialization requires calling back
+        // out to Python.
+	PyThreadState *_save = nullptr;
```
Comment (nwhitehorn) [inline core/src/G3Writer.cxx:41]: It looks like you have some mixed tabs and spaces here. Can you please standardize on tabs?

File: core/src/G3Writer.cxx [hunk 2]
```diff
@@ -34,6 +34,16 @@ G3Writer::~G3Writer()
...
+	if (Py_IsInitialized() && PyGILState_Check()) {
+		frame->GenerateBlobs();
```
Comment (nwhitehorn) [inline core/src/G3Writer.cxx:43]: There's no actual harm to calling GenerateBlobs() unconditionally if you want to simplify this conditional (also no benefit really).

**Review by nwhitehorn — CHANGES_REQUESTED** [review]
Thanks! Some minor comments in the diff. This is known to solve the SMuRF issue?

**Comment by mhasself** [comment]
Fixes as per your comments.  Jack says he can try this on the Smurf server today, and we'll see if it helps.  Indeed it's not clear if our main difficulty is IO or CPU.

**Review by nwhitehorn — APPROVED** [review]
Looks to me! Feel free to merge as soon as you know it fixes the problem you have.

**Pull request merged**